### ISSUE TITLE
reinstalling the linux agent on Ubuntu / Debian

### ISSUE THREAD

**Issue by danielsollondon** [Azure/WALinuxAgent#1142]
During testing on Ubuntu 16.04, I provision a ‘custom image’ with only cloud-init, then I install the Linux Agent, but because Debian packages have a behavior of autostart at package install, this then causes the Linux Agent to go through the provisioning workflow, and reset the SSH keys. This result is if you ssh out and ssh in, you see a man in the middle ssh error.

Is this something that we need to resolve in the linux agent?

**Comment by hglkrijger** [comment]
I think the implied question here is whether there should continue to be a single PA+GA or whether we should split these into two. Adding in @boumenot and @paulmey for their thoughts.

If we decide to keep a single package, we would need some way to distinguish between 'install for provisioning' and 'install for extension handling'.

**Comment by paulmey** [comment]
Funny enough I was thinking about this yesterday... I think splitting makes a lot of sense. Aside from being able to provide sensible defaults for the 'enabled' state of the services, it would also allow you to declarative wait for provisioning to complete in systemd by waiting on the service.

**Comment by narrieta** [comment]
We have a flag to disable the provision part of the agent. This is described in our [wiki](https://github.com/Azure/WALinuxAgent/wiki/VMs-without-WALinuxAgent).

Closing this issue since that flag seems to address the problem described here. Please re-open if that is not the case.
